src/main/java/de/tum/in/www1/artemis/web/rest/LearningPathResource.java

```diff
@@ -399,11 +401,52 @@ public ResponseEntity<List<LearningPathNavigationObjectDTO>> getLearningObjectsF
-                .map(learningObject -> LearningPathNavigationObjectDTO.of(learningObject, learningObjectService.isCompletedByUser(learningObject, user), competencyId)).toList();
+        List<LearningPathNavigationObjectDTO> learningObjects = learningPathRecommendationService.getOrderOfLearningObjectsForCompetency(competencyId, user, learningPathId)
+                .stream().map(learningObject -> LearningPathNavigationObjectDTO.of(learningObject, learningObjectService.isCompletedByUser(learningObject, user), competencyId))
+                .toList();
         return ResponseEntity.ok(learningObjects);
     }
 
+    /**
+     * GET courses/:courseId/learning-paths/configuration : Gets the learning paths configuration for the course
+     *
+     * @param courseId the id of the course for which to get the learning paths configuration
+     * @return the ResponseEntity with status 200 (OK) and with the configuration in the body
+     */
+    @GetMapping("courses/{courseId}/learning-paths/configuration")
+    @FeatureToggle(Feature.LearningPaths)
+    @EnforceAtLeastInstructorInCourse
+    public ResponseEntity<LearningPathsConfigurationDTO> getLearningPathsConfiguration(@PathVariable long courseId) {
+        log.debug("REST request to get learning paths configuration for course with id: {}", courseId);
+        Course course = courseRepository.findWithEagerLearningPathsConfigurationByIdElseThrow(courseId);
+        if (!course.getLearningPathsEnabled()) {
+            throw new BadRequestException("Learning paths are not enabled for this course.");
+        }
+
+        return ResponseEntity.ok(LearningPathsConfigurationDTO.of(course.getLearningPathsConfiguration()));
+    }
+
+    /**
+     * PUT courses/:courseId/learning-paths/configuration : Updates the learning paths configuration for the course
+     *
+     * @param courseId                   the id of the course for which to update the learning paths configuration
+     * @param learningPathsConfiguration the updated configuration
+     * @return the ResponseEntity with status 200 (OK)
+     */
+    @PutMapping("courses/{courseId}/learning-paths/configuration")
+    @FeatureToggle(Feature.LearningPaths)
+    @EnforceAtLeastInstructorInCourse
+    public ResponseEntity<Void> updateLearningPathsConfiguration(@PathVariable long courseId, @RequestBody LearningPathsConfigurationDTO learningPathsConfiguration) {
+        log.debug("REST request to update learning paths configuration for course with id: {}", courseId);
+        Course course = courseRepository.findWithEagerLearningPathsConfigurationByIdElseThrow(courseId);
+        if (!course.getLearningPathsEnabled()) {
+            throw new BadRequestException("Learning paths are not enabled for this course.");
+        }
+        learningPathService.changeLearningPathsConfiguration(course, learningPathsConfiguration);
+
+        return ResponseEntity.noContent().build();
```

Comment:
The comment says we have status `200 OK` on success, but we return `204 No Content`. Can you either update the comment & server test to refer to/expect 204 or change the return value of this method to `ResponseEntity.ok`?